From 7d5e1b8ce5abe5a640005948d09262394df2b3a7 Mon Sep 17 00:00:00 2001
From: Sameer Mulla <sameer.mulla@lge.com>
Date: Wed, 12 Jun 2019 11:42:52 +0530
Subject: [PATCH] Fix add service failure

:Release Notes:
Ignore Includes property  even parsing is failed

:Detailed Notes:
Include parameter is optional so even parsing fails
ignore includes

:Testing Performed:
Builded and tested

:QA Notes:

:Issues Addressed:
[PLAT-79811] [webOS OSE]: GATT add service failed.

Upstream-Status: Inappropriate[webos specific]
---
 src/gatt-database.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/gatt-database.c b/src/gatt-database.c
index 99c8ea2..93434e0 100644
--- a/src/gatt-database.c
+++ b/src/gatt-database.c
@@ -1788,20 +1788,20 @@ static bool parse_includes(GDBusProxy *proxy, struct external_service *service)
 		return true;
 
 	if (dbus_message_iter_get_arg_type(&iter) != DBUS_TYPE_ARRAY)
-		return false;
+		return true;
 
 	dbus_message_iter_recurse(&iter, &array);
 
 	do {
 		if (dbus_message_iter_get_arg_type(&array) !=
 						DBUS_TYPE_OBJECT_PATH)
-			return false;
+			return true;
 
 		dbus_message_iter_get_basic(&array, &obj);
 
 		if (!queue_push_tail(service->includes, obj)) {
 			error("Failed to add Includes path in queue\n");
-			return false;
+			return true;
 
 		}
 
-- 
2.7.4

